### PR TITLE
Create Docker instruction

### PR DESCRIPTION
**Pull request by Dmitriy-DevOps-trainee**

### CODE REVIEW

**Comment by Dmitriy-DevOps-trainee** [comment]
@cakemanator
@yegorgeorgiyev
@shlepanets
@nikita-safonov

File: Docker instruction [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Pull this files 
+docker pull dimitrii89/headnode:headnode
+docker pull dimitrii89/worker:worker
+
+# Open the ports: 8088, 9870
+
+# After downloading images. Follow this steps:
+sudo docker network create hadoop-net
+sudo docker build -f Dockerfile -t hadoop-headnode:1.0 .
```
Comment (Cakemanator) [inline Docker instruction]: Cannot find the Dockerfiles

**Review by Cakemanator — CHANGES_REQUESTED** [review]

**Review by Cakemanator — CHANGES_REQUESTED** [review]
Changes need to be done

File: Docker instruction [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+# Dockerfiles:
+
+# Dockerfile_hn
+
+FROM centos:7
+
+MAINTAINER Dima <dimitry.vlasov@gmail.com>
+
+ARG HADOOP_HOME="/usr/local/hadoop/current"
+ENV HADOOP_HOME="/usr/local/hadoop/current"
```
Comment (Cakemanator) [inline Docker instruction]: Why define that 2 times?

File: Docker instruction [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    sed -i "s/%DATANODE_DIRS%/\/opt\/mount1\/datanode-dir,\/opt\/mount2\/datanode-dir/g" /usr/local/hadoop/current/etc/hadoop/hdfs-site.xml && \
+    sed -i "s/%NODE_MANAGER_LOCAL_DIR%/\/opt\/mount1\/nodemanager-local-dir,\/opt\/mount2\/nodemanager-local-dir/g" /usr/local/hadoop/current/etc/hadoop/yarn-site.xml && \
+    sed -i "s/%NODE_MANAGER_LOG_DIR%/\/opt\/mount1\/nodemanager-log-dir,\/opt\/mount2\/nodemanager-log-dir/g" /usr/local/hadoop/current/etc/hadoop/yarn-site.xml && \
+    sed -i "s/host1/0.0.0.0/g" /usr/local/hadoop/current/etc/hadoop/yarn-site.xml
+
+RUN mkdir /usr/local/hadoop/current/logs && \
+    chown hdfs:hadoop /usr/local/hadoop/current/logs && \
+    chmod g+w /usr/local/hadoop/current/logs
+
+RUN  mkdir -p /opt/mount{1,2}/namenode-dir && \
```
Comment (Cakemanator) [inline Docker instruction]: Can we optimize the amount of `RUN` instructions? What will it give to us?

File: Docker instruction [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+     chown yarn:hadoop /opt/mount2/nodemanager-local-dir && \
+     chown yarn:hadoop /opt/mount1/nodemanager-log-dir && \
+     chown yarn:hadoop /opt/mount2/nodemanager-log-dir
+
+VOLUME ["/opt/mount1", "/opt/mount2"]
+
+COPY start_wn.sh /opt/start.sh
+
+USER root
+CMD ["/bin/bash", "/opt/start.sh"]
```
Comment (Cakemanator) [inline Docker instruction:115]: Same for this dockerfile

**Review by Cakemanator — CHANGES_REQUESTED** [review]

File: Docker instruction (hunk 2 above)
Comment (Dmitriy-DevOps-trainee) [reply Docker instruction]: ARG variables can be used to set default settings for ENV variables from the command line during the image build process.
That's why I wrote these lines. And it works. I didn't find any mistakes when creating images and then checking the browser.

File: Docker instruction (hunk 3 above)
Comment (Dmitriy-DevOps-trainee) [reply Docker instruction]: You mean to reduce the number of command RUN.  I'm sure we can. I think that this optimization will affect the performance. That's my first experience in Docker and I'm trying to follow step by step, layer by layer.

File: Docker instruction (hunk 4 above)
Comment (Dmitriy-DevOps-trainee) [reply Docker instruction:115]: Check prevoius comments.

File: Docker instruction (hunk 2 above)
Comment (Cakemanator) [reply Docker instruction]: But there is no connections between `ARG` and `ENV`.
If you want to set `ENV` from `ARG` then:
```
ARG HADOOP_HOME="/usr/local/hadoop/current"
ENV HADOOP_HOME="$HADOOP_HOME"
```

File: Docker instruction (hunk 3 above)
Comment (Cakemanator) [reply Docker instruction]: Not only performance but image size as well.
Please do that (I mean `RUN` optimization)

**Review by Cakemanator — APPROVED** [review]

**Pull request merged**